scheduler_test.go

```diff
@@ -535,38 +422,750 @@ func TestScheduler_Stop(t *testing.T) {
 	})
 }
 
-func TestMonths(t *testing.T) {
-	scheduler := NewScheduler(time.Local)
-	now := time.Now()
-	scheduler.Every(1).Month(now.Day()).Do(func() {})
-	_, nextRun := scheduler.NextRun()
-	assert.Equal(
-		t,
-		time.Date(now.Year(), (now.Month()+time.Month(1))%12, now.Day(), 0, 0, 0, 0, time.Local),
-		nextRun,
-	)
-}
-
 func TestScheduler_StartAt(t *testing.T) {
 	scheduler := NewScheduler(time.Local)
 	now := time.Now()
 
 	// With StartAt
 	job, _ := scheduler.Every(3).Seconds().StartAt(now.Add(time.Second * 5)).Do(func() {})
+	scheduler.scheduleAllJobs()
 	_, nextRun := scheduler.NextRun()
-	assert.Equal(
-		t,
-		now.Add(time.Second*5),
-		nextRun,
-	)
+	t.Log("now: " + now.String())
+	t.Log("next run : " + nextRun.String())
+	t.Log("now plus 5 run : " + now.Add(time.Second*5).String())
+	assert.Equal(t, now.Add(time.Second*5), nextRun)
 	scheduler.Remove(job)
 
 	// Without StartAt
-	scheduler.Every(3).Seconds().Do(func() {})
+	job, _ = scheduler.Every(3).Seconds().Do(func() {})
+	scheduler.scheduleNextRun(job)
 	_, nextRun = scheduler.NextRun()
-	assert.Equal(
-		t,
-		now.Add(time.Second*3).Second(),
-		nextRun.Second(),
-	)
+	assert.Equal(t, now.Add(time.Second*3).Second(), nextRun.Second())
+}
+
+func TestScheduler_ScheduleDays(t *testing.T) {
+	sched := NewScheduler(time.UTC)
+	fakeTime := fakeTime{}
+	sched.time = &fakeTime
+
+	t.Run("schedule for days without .At(time)", func(t *testing.T) {
```

Comment:
the t.Run blocks look to be repeated, could make this a [table test](https://github.com/golang/go/wiki/TableDrivenTests)?